app/config/services/platform.xml

```diff
@@ -422,16 +422,42 @@
         </service>
 
         <!-- Search -->
-        <service id="app.search.search_results_filter" class="AppBundle\Search\SearchParametersFilter">
+        <service id="AppBundle\Search\SearchParametersFilter" shared="false">
```

Comment:
I fail to see when do we need to have it instantiated many times?

Reply:
Yes normal, I did like this because, in theory, each object is like an entity with its own values.

Reply:
It depends how you use these 2 instances. At the very beginning yes but then their values can  change independently.

Reply:
So you're saying we have some stageful services?

Reply:
What do you mean ?

Reply:
I mean services should be stateless (sans état), and their configuration should not change during runtime to avoid undesired side effects.

>It depends how you use these 2 instances. At the very beginning yes but then their values can change independently.

I still got two questions:
 - when does it happens to handle 2 instances in the same runtime?
 - how can their values changes during runtime?

Reply:
So yea it's stateful because the object is meant to be changed at runtime.

- When does it happens to handle 2 instances in the same runtime?
Never at the moment. Like I said, I did that from a theoretical point of view

- how can their values change during runtime?
At the moment, the value changed based on how you use setter methods and the HTTP request (see `handleRequest`).

Reply:
I'm sorry I cannot see any reason to perform 2 search queries with the same Request. Calling the setter for maxResults does not seem necessary we could inject it in the constructor and use the default constant if no value is passed. The setter should only be used with a from type mapping attributes from the request.

Do I miss something?

Reply:
Yet another question, do we really need a parameter hardcoded in the config file AND a default class constant, both using the same value... ?

Reply:
> I'm sorry I cannot see any reason to perform 2 search queries with the same Request. Calling the setter for maxResults does not seem necessary we could inject it in the constructor and use the default constant if no value is passed. The setter should only be used with a from type mapping attributes from the request.

I know and I agree but I cannot rewrite the whole feature in this PR.

> Yet another question, do we really need a parameter hardcoded in the config file AND a default class constant, both using the same value... ?

Good question. I don't know why it was done like this. I guess it's to provide more flexibility just in case. Right now it looks useless.

Reply:
Ok sorry again because I review an already implemented feature, and understand this is out of your scope. Now this is good to have this conversation on record.

Finally I suggest to use a shared service anyway.

Reply:
You are very welcome ! Thank you for your review 👍